trainers/loss_fn.py

```diff
@@ -75,7 +75,7 @@ def compute_perplexity(logits, y, char_lengths, mask=None):
     # flatten both
     logits = logits.view(-1, logits.size(-1))
     y = y.view(-1)
-    loss = torch.nn.functional.cross_entropy(logits, y, reduction="none")
+    loss = torch.nn.functional.cross_entropy(logits, y, reduction="none", ignore_index=-1)
```

Comment:
we are 100% sure this has no impacts elsewhere? should be okay but..

Reply:
Yes. Ignore_index's default value is -100. So, changing it from -100 to -1 would only impact existing label (y) assignments of -1 or -100. 

1) I searched our code base for the assignment of -100, there was none. 
2) I searched our code base for the assignment of -1. Apart from the new `MLMDataLoader` using label[~mask]=-1, other assignments were used in arguments such as 'dim=-1', or just as index values.

Reply:
![image](https://github.com/LeonGuertler/SuperTinyLanguageModels/assets/95431329/b71fd81a-abb5-4cc7-9330-56920880048c)

Reply:
![image](https://github.com/LeonGuertler/SuperTinyLanguageModels/assets/95431329/e339e269-e7a2-4c99-8287-eef4c94414a9)